Suggested Readme Edits (#1)
uwblueprint/bootcamp-mern-rest: README.md

Keep in mind that both the frontend and backend have hot reload enabled so you will rarely need to restart the application.

## Warm Up

**Skip this if you are familiar with how to make HTTP Requests**

Before starting the tasks, let's try adding an entry to the existing Restaurant table. 

If we look at `/backend/routes/restaurant.js`, we can see each endpoint defined as its own function.

> Reading the comments, if we want to create a restaurant, we will need to trigger the code in `router.post("/", ...)` by making a `POST` request to the *path* `/`.  

Reviewer: ```suggestion
Reading the comments, if we want to create a restaurant, we will need to trigger the code in `router.post("/", ...)` by making a `POST` request to the *path* `/`. Note the path is relative to the root URL that we configured for this router in `server.js`, so the full path is actually `/api/restaurants/`. We can usually omit trailing "/"s in URLs.
```

If you look in the file `server.js`, you see that `app.listen(5000, ...)` tells us the server is listening for requests on *port* `5000`.

> Since we are running this *locally*, we can put all these peices of information together into one url: `http://localhost:5000/` .

Reviewer: ```suggestion
Since we are running this *locally*, we can put all these pieces of information together into one url: `http://localhost:5000/api/restaurants` .
```

> Try downloading [Postman](https://www.postman.com/), and make a `POST` request to `http://localhost:5000/`. You can read the *Tip* section below or the documentation to learn more about Postman.

Reviewer: ```suggestion
Try downloading [Postman](https://www.postman.com/), and make a `POST` request to `http://localhost:5000/api/restaurants`. You can read the *Tip* section below or the documentation to learn more about Postman.
```

Following the starter-code's pattern, complete these tasks:

Currently, our restaurant directory maintains a singular list of restaurants. 
Suppose we now want the ability to create restaurant groups (e.g. "Uni Plaza", "Cafes", "Open Late", etc.). 

A single group can contain multiple restaurants, and a single restaurant can be part of multiple groups.

A `RestaurantGroup` has these fields: `id`, `name`, `description`, `restaurantIds`

> 1. Using the existing code as a template, create REST endpoints for `RestaurantGroup`, supporting C.R.U. (Create, Retrieve, Update) operations. 

Reviewer: Look like update got moved to Extensions?

Author: Good catch!

>     a. To support _creating_ groups, make an endpoint called `/groups` which handles a `POST` request. It will accept a `name`, `description`, and `restaurantIds` as the request parameters to create a new Restaurant Group. It will return an `id` which corresponds to the newly created Restaurant Group if successful

Reviewer: > request parameters

I think request body? Might help if we include the expected request and response JSON structures here 🙂

Author: Good catch! And yeah good idea I'll add that. Were you thinking of the ids just being an array?

Reviewer: Yep! Array of Mongo objectIds, e.g. `[5ff7e50a3e569d00810b67f5, 5ff7e50a3e569d00810b67f6]`

    b. To support _retrieving_ a group, make an endpoint called `/groups` which handles a `GET` request. It will accept an `id` and return the Restaurant Group data corresponding to that `id`
    
2. Display `RestaurantGroup` data in the frontend (try to reuse existing components to save time, don't worry about design and appearance)

3. Modify the API `GET` response so that it includes full restaurant information, not just the ids

### Tip

For some help with debugging your work, you can download [Postman](https://www.postman.com/)! Postman is a tool which allows developers to test endpoints with different requests, and view its responses. This will allow you to write and test the backend code without making any changes to the frontend yet.

And please **ASK FOR HELP**, if you're stuck!

> ### Extensions

Author: I was thinking we could break up the assignment in sections so that it was less overwhelming for new developers.
